src/main.rs

```diff
@@ -232,76 +299,48 @@ fn main() {
 
     // We can only release, if we are allowed to write
     let release_mode = write_mode && string_to_bool(&args.flag_release);
+    let repository_path = get_repository_path(&args);
 
     config_builder.write(write_mode);
     config_builder.release(release_mode);
-    config_builder.branch(args.flag_branch);
-
-    println!("semantic.rs 🚀");
-
-    logger::stdout("Analyzing your repository");
-    let path = Path::new(&args.flag_path);
-    let path = fs::canonicalize(path)
-        .unwrap_or_else(|_| print_exit!("Path does not exist or a component is not a directory"));
-    let repository_path = path.to_str()
-        .unwrap_or_else(|| print_exit!("Path is not valid unicode"));
-
-    let repo = match git2::Repository::open(repository_path) {
-        Ok(repo) => repo,
-        Err(e) => {
-            logger::stderr(format!("Could not open the git repository: {:?}", e));
-            process::exit(1);
-        }
-    };
-
-    config_builder.repository_path(repository_path.to_owned());
-
-    // extra scope scope to make sure borrow of `repo` is dropped
-    {
-        let signature = match git::get_signature(&repo) {
-            Ok(sig) => sig,
-            Err(e) => {
-                logger::stderr(format!("Failed to get the committer's name and email address: {}", e.description()));
-                logger::stderr(COMMITTER_ERROR_MESSAGE);
-                process::exit(1);
-            }
-        };
+    config_builder.branch(args.flag_branch.clone());
+    config_builder.repository_path(repository_path.clone());
+    config_builder.signature(get_signature(repository_path.clone()));
+    let (user, repo) = get_user_and_repo(&repository_path);
+    if user.is_some() {
```

Comment:
If we change the return type of `get_user_and_repo`, this would be easy to turn into an if-let statement:

```rust
if let Some(user, repo) = get_user_and_repo(&repository_path) {
...
}
```